packages/flutter_tools/gradle/aar_init_script.gradle

```diff
@@ -42,17 +42,11 @@ void configureProject(Project project, File outputDir) {
     // Check if the project uses the Flutter plugin (defined in flutter.gradle).
     Boolean usesFlutterPlugin = project.plugins.find { it.class.name == "FlutterPlugin" } != null
     if (!usesFlutterPlugin) {
-        // Plugins don't include their dependencies under the assumption that the parent project adds them.
-        if (project.properties['android.useAndroidX']) {
-            project.dependencies {
-                compileOnly "androidx.annotation:annotation:+"
-            }
-        } else {
-            project.dependencies {
-                compileOnly "com.android.support:support-annotations:+"
-            }
-        }
         project.dependencies {
+            // Some plugins don't include `annotations` and they don't set
+            // `android.useAndroidX=true` in `gradle.properties`.
+            compileOnly "androidx.annotation:annotation:+"
```

Comment:
What happens if this script is ran against an app that uses the support libraries and hasn't migrated to AndroidX/isn't using Jetifier?

Reply:
this won't affect the output artifact "AAR" since it's a `compileOnly` dependency.  My understanding is that annotations are just hints, and any reference in code is removed and it's not included in either the output `.jar` or `.aar`.